Dockerfile

```diff
@@ -0,0 +1,132 @@
+# Based on https://github.com/ipfs/go-ipfs/tree/c55fda4d6cbd40150d4432ea9226c5b977fc96f4
+FROM golang:1.16.7-buster as clone
+
+WORKDIR /clone
+
+RUN git clone --depth 1 --branch v0.10.0 https://github.com/ipfs/go-ipfs
+
+# Note: when updating the go minor version here, also update the go-channel in snap/snapcraft.yml
+FROM golang:1.16.7-buster
+
+# Install deps
+RUN apt-get update && apt-get install -y \
+  libssl-dev \
+  ca-certificates \
+  fuse
+
+ENV SRC_DIR /go-ipfs
+
+# Download packages first so they can be cached.
+COPY --from=clone /clone/go-ipfs/go.mod /clone/go-ipfs/go.sum $SRC_DIR/
+COPY --from=clone /clone/go-ipfs $SRC_DIR
+
+# RUN cd $SRC_DIR \
+#   && go mod edit -replace \
+#   github.com/ipfs/go-ds-s3=github.com/obo20/go-ds-s3@v0.7.0
+
+RUN cd $SRC_DIR \
```

Comment:
It's mainly just these lines you've added to the regular `go-ipfs` Dockerfile, right?

```
RUN cd $SRC_DIR \
  && go get github.com/3box/go-ds-s3/plugin@v0.7.0-test-1 \
  && go get github.com/cheggaaa/pb@v1.0.29

RUN cd $SRC_DIR \
  && echo "\ns3ds github.com/3box/go-ds-s3/plugin 0" >> plugin/loader/preload_list
```

If so, what would you think about doing something like [this](https://github.com/ceramicnetwork/js-ceramic/blob/develop/.github/workflows/debug-env.yml#L26) to inject your commands into the regular Dockerfile. 

Not foolproof but might help avoid keeping up with Dockerfile changes in the main `go-ipfs` repo.